text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,405 @@
+
+Once the _Developer Preview_ has been declared, breaking changes will only be allowed if they are motivated by
+overwhelming customer feedback or the absence of alternate solutions to fix a well-documented pain point introduced by
+v2 changes.
+
+## Feedback Phase
+
+During _Developer Preview_, we will keep listening to customer feedback and address bugs as they are discovered. The CDK
+[version reporting] dataset will be used to track adoption of the new version. We will encourage customers to provide
+feedback on the upcoming new release using a pinned GitHub issue, and possibly other means.
+
+[version reporting]: https://docs.aws.amazon.com/cdk/latest/guide/tools.html
+
+## General Availability
+
+After the _Developer Preview_ has been sufficiently vetted, a timeline for _General Availability_ will be agreed upon by
+maintainers. The _General Availability_ checkist includes the following items:
+
+- [ ] Documentation
+  - [ ] API Reference Documentation
+  - [ ] User Guide
+  - [ ] Migration Guide
+  - [ ] Final maintenance plan for v1
+- [ ] Quality Assurance
+  - [ ] Complete suite of integration tests
+  - [ ] Examples built and tested against v2
+- [ ] No v1 feature is missing from v2 (unless it was deprecated in v1)
+
+Once the criteria have been satisfied, the high-level procedure for officially releasing CDK v2 is:
+
+1. Create a new `v1` branch from the `master` branch
+   1. Configure GitHub branch protection on `v1` as appropriate
+   1. Deploy CI/CD infrastructure on top of `v1`
+1. Make `next` the new `master` branch:
+   1. Force-push `next` on `master`
+   1. Delete `next` and all related CI/CD infrastructure
+1. Perform a _normal_ release
+1. Ceremoniously annouce CDK v2 is _Generally Available_, and repeat the v1 maintenance plan
+
+## Clean Up
+
+Now that the new major version has been declared _Generally Available_, the previous major (`v1`) has entered
+_maintenance_ mode. Forward porting is no longer exercised: instead, bug fixes on `master` will be back-ported to `v1`
+when necessary. This means `master` can diverge more significantly from `v1` than in earler stages of the process.
+
+The definition of `v1` feature flags and the code paths supporthing the _old behavior_ of those can be deleted.
+
+## Monitoring Adoption
+
+We will keep an eye on adoption of CDK v2 thanks to the [version reporting] dataset. Should adoption pick up too slowly,
+actions will be taken to understand the causes for slow adoption and identify opportunities to provide additional
+tooling to remove roadblocks and facilitate migration.
+
+[version reporting]: https://docs.aws.amazon.com/cdk/latest/guide/tools.html
+
+## Deprecation of v1
+
+Based on the v1 maintenance plan, feature development is to stop on CDK v1, as only bugs reported against CDK v1 are to
+be fixed. Back-porting of features is not necessary, although the community is still welcome to back-port features.
```

Comment:
Is this the plan? To fix all bugs reported against 1.x both on 1.x and on 2.x? That's going to be pretty expensive. This is something we need to be very intentional about.

Reply:
Either the bug exists in v1 AND v2, and then the fix is likely the same on both sides. Ergo I am assuming it will not be "significantly" more expensive to fix both.

Or the bug exists only in v1 and there is nothing to fix in v2.